internal/http/services/owncloud/ocdav/delete.go

```diff
@@ -137,8 +137,8 @@ func (s *svc) handleSpacesDelete(w http.ResponseWriter, r *http.Request, spaceID
 
 	// do not allow deleting spaces via dav endpoint - use graph endpoint instead
 	// we get a relative reference coming from the space root
-	// so if the path is "empty" we a referencing the space
-	if ref.GetPath() == "." {
+	// so if the path is "empty" and no opaque id is present, we are referencing the space
+	if ref.GetPath() == "." && ref.GetResourceId().GetOpaqueId() == "" {
```

Comment:
The check is not correct. It could also be a space root if
```
ref.GetResourceId().GetOpaqueId() == ref.GetResourceId().GetSpaceId() && ref.GetPath() == "."
```

Reply:
True! But we still need to check for `ref.GetResourceId().GetOpaqueId() == ""`, no? Because we can't say for sure that the opaqueId is given?

Reply:
Yes exactly. We need to check both. Good luck with making the condition readable :smile:

Reply:
Gave it my best shot 😄